components/qma6100p/qma6100p.c

```diff
@@ -0,0 +1,471 @@
+/*
+ * SPDX-FileCopyrightText: 2024 Espressif Systems (Shanghai) CO LTD
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <math.h>
+#include <time.h>
+#include <sys/time.h>
+#include "esp_system.h"
+#include "driver/i2c.h"
```

Comment:
Leaving this here for discussion:
Do we want to use the new Driver-NG here? with `"driver/i2c_master.h"`?

cc @espzav